nni/algorithms/compression/v2/pytorch/pruning/tools/metrics_calculator.py

```diff
@@ -19,7 +19,7 @@ class StraightMetricsCalculator(MetricsCalculator):
     def calculate_metrics(self, data: Dict[str, Tensor]) -> Dict[str, Tensor]:
         metrics = {}
         for name, tensor in data.items():
-            metrics[name] = tensor.clone().detach()
+            metrics[name] = tensor.clone().detach_()
```

Comment:
Could you please add more comments to explain why `detach_` rather that `detach`?

Reply:
add comments, in fact, just avoid creating a new tensor